pytorch_lightning/trainer/training_loop.py

```diff
@@ -200,10 +200,10 @@ def on_train_batch_end(self, epoch_output, batch_end_outputs, batch, batch_idx,
         self.trainer.logger_connector.on_train_batch_end()
 
     def reset_train_val_dataloaders(self, model):
-        if self.trainer.train_dataloader is None or not self.trainer.reload_dataloaders_every_epoch:
```

Comment:
Can you add a test to make sure different dataloaders get properly reloaded when reload_dataloaders_every_epoch=True ?

Reply:
could you describe what's missing from the tests here? https://github.com/PyTorchLightning/pytorch-lightning/blob/ac7d6a35c3d36267b1f0f66ee6609ac40d244dba/tests/trainer/test_dataloaders.py#L1201